linker_script_rz_a1l.ld

```diff
@@ -186,32 +186,27 @@ SECTIONS
 		*(.data.*)
 		_edata = .;
 		
-		
-		/* Uncommenting this section back in allows the section at the top of the main C++ file to be uncommented...
-		But I'm not sure if this is really the right place for this. */
-		
-		/*
-		__CTOR_LIST__ = .;
-		. = ALIGN(2);
-		__ctors = .;
-		*(.ctors)
-		__ctors_end = .;
-		__CTOR_END__ = .;
-		__DTOR_LIST__ = .;
-		___dtors = .;
-		*(.dtors)
-		___dtors_end = .;
-		__DTOR_END__ = .;
-		. = ALIGN(2);
-		_mdata = .;
-		*/
-		
 		*(.got.plt) 
 		*(.got)
 		.	= ALIGN(0x8);
 		address_end_data_ROM = .;
 	} > RAM012L
-	
+
```

Comment:
I think you'll also need a `.preinit_array` section in here to deal with vtable stuff (validation and maybe init?)

References: 
[Picolibc docs (but also applies for newlib)](https://github.com/picolibc/picolibc/blob/main/doc/init.md)
[Info on the SysV ABI](https://stackoverflow.com/a/60178049)

Reply:
thanks, done